server/src/main/java/org/candlepin/model/Product.java

```diff
@@ -292,6 +304,13 @@ public Product(String productId, String name, String variant, String version, St
         this.setAttribute(Attributes.ARCHITECTURE, arch);
     }
 
+    public Product(String productId, String name, String variant, String version, String arch, String type,
```

Comment:
I commented in the other PR that builds off this, but this constructor isn't used anywhere from what I can see, and we should try to avoid adding extra constructors and encourage long-form object construction with fluent-style interfaces.

Reply:
Sure, apparently I was using it in a unit test, which is now not relevant. Removing it from here.